test/functional/wallet_multiwallet.py

```diff
@@ -129,15 +126,33 @@ def wallet_file(name):
         for wallet_name in to_load:
             self.nodes[0].loadwallet(wallet_name)
 
-        os.mkdir(wallet_dir('no_access'))
-        os.chmod(wallet_dir('no_access'), 0)
-        try:
+        # Tests for possible scanning errors:
+        # 0. Baseline, no errors.
+        with self.nodes[0].assert_debug_log(expected_msgs=[], unexpected_msgs=['Error scanning']):
+            walletlist = self.nodes[0].listwalletdir()['wallets']
+        assert_equal(sorted(map(lambda w: w['name'], walletlist)), sorted(in_wallet_dir))
+        # 1. "Permission denied" error.
+        if platform.system() != 'Windows' and os.geteuid() != 0:
+            os.mkdir(wallet_dir('no_access'))
+            os.chmod(wallet_dir('no_access'), 0)
+            try:
+                with self.nodes[0].assert_debug_log(expected_msgs=['Error scanning']):
+                    walletlist = self.nodes[0].listwalletdir()['wallets']
+            finally:
+                # Need to ensure access is restored for cleanup
+                os.chmod(wallet_dir('no_access'), stat.S_IRUSR | stat.S_IWUSR | stat.S_IXUSR)
+            assert_equal(sorted(map(lambda w: w['name'], walletlist)), sorted(in_wallet_dir))
+        # 2. "Too many levels of symbolic links" error.
+        # This test cannot be conducted robustly on Windows
+        # because it depends on the build toolchain:
```

Comment:
In c0127439597ee9d09b8e117be7d6226a68b45721:

> because it depends on the build toolchain:

Even if this is the case, is the best option to just to do nothing? Shouldn't we atleast test the known behaviour for our release build, so we'd catch if this changes (is fixed?) in future? Otherwise my understand is if this changes again, it's just going to be hidden/ignored.

Reply:
> I guess the check could be limited to exclude the msvc build?

That would make the most sense to me.

Reply:
> Therefore, the release build won't trigger the "Error scanning" message, and this test would fail on the master branch.

Sure, so you can check that that is happening when testing on Windows.

Reply:
> I guess the check could be limited to exclude the msvc build?

Unfortunately, I can't find a simple way to detect whether binaries were built with MSVC without adding extra complexity to the Python code.